src/new/wallet.js

```diff
@@ -1106,12 +1169,28 @@ class HathorWallet extends EventEmitter {
     let wsData = this.wsTxQueue.dequeue();
 
     while (wsData !== undefined) {
-      // process wsData like it just arrived
+      // save new txdata
       await this.onNewTx(wsData);
       wsData = this.wsTxQueue.dequeue();
       // We should release the event loop for other threads
-      // This effectively awaits 0 seconds, but it schedule the next iteration to run after other threads.
-      await new Promise(resolve => { setTimeout(resolve, 0) });
+      // This effectively awaits 0 seconds
+      // but it schedule the next iteration to run after other threads.
+      await new Promise(resolve => { setTimeout(resolve, 0); });
+    }
+
+    await this.storage.processHistory();
+  }
+
+  async checkScanningPolicy() {
+    // check address scanning policy and load more addresses if needed
+    const loadMoreAddresses = await checkScanningPolicy(this.storage);
```

Comment:
question: I didn't understand this recursion. Is there another method with this same name?

Reply:
nitpick(non-blocking): I still think it's a bit confusing. A way to explicit this is being called on the storage would help here.

Just a comment I believe would be enough to avoid this confusion, but another more complex alternative would be importing `storageUtils` as a dedicated variable and calling `storageUtils.checkScanningPolicy`.

Reply:
Changed the name of the method to `scanAdressesToLoad` it will use the scanning policy util to determine which addresses need to be loaded and sync the history of them.

Also added a docstring to the method.